### PR TITLE
[MacOS] Handle MediaKey callback in main thread

### LINKED ISSUE #24971: v21.0 crash when using volume control or other special keys
### Describe the bug
On a Mac, pressing any special key (volume control, brightness, playback control, keyboard backlight etc.) causes Kodi to crash immediately. Other keys are working just fine.

## Expected Behavior
Don't crash when turning the volume up or down.

## Actual Behavior
Crash.

## Possible Fix
Rolling back to v20.5 fixes the problem

### To Reproduce
1. Install Kodi v21.0
2. Launch it
3. Press volume up (or any other service key)

### Debuglog

The debuglog can be found here:
Kodi log: https://pastebin.com/5hgd9Saj
macOS crash log: https://pastebin.com/uhQh4b4p

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [x] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: macOS Catalina 10.15.7
 - Kodi version: 21.0

### PR DESCRIPTION
**Pull request by enen92**
## Description
This PR makes sure the callback event for media keys runs on the main thread similar to other input events. Fixes a crash if somehow the thread responsible for the dispatch of the event is destructed in between the actual process of the key event.

## Motivation and context
Fixes https://github.com/xbmc/xbmc/issues/24971

## How has this been tested?
Runtime tested, confirmed fixed by the issue reporter in the issue thread

## What is the effect on users?
Nothing special, avoid the crash

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

**Comment by enen92** [comment]
@kambala-decapitator looks like this was the reason for the try/catch in VLC: https://github.com/videolan/vlc/blob/f7bb59d9f51cc10b25ff86d34a3eff744e60c46e/modules/gui/macosx/imported/SPMediaKeyTap/SPMediaKeyTap.m#L270-L277

However this is wrong, according to the docs and as you pointed out on your review this is not supposed to throw. So it's a bandaid for the actual problem.

File: xbmc/platform/darwin/osx/MediaKeys.mm [hunk 1]
```diff
@@ -16,25 +16,33 @@
 #include "utils/log.h"
 
 #import <AppKit/AppKit.h>
+#import <Foundation/Foundation.h>
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/MediaKeys.mm:19]: Not sure what this is for. GCD facilities are in dispatch.h iirc

**Review by kambala-decapitator — APPROVED** [review]
The change itself looks good

File: xbmc/platform/darwin/osx/MediaKeys.mm (hunk 1 above)
Comment (enen92) [reply xbmc/platform/darwin/osx/MediaKeys.mm:19]: This was because of dispatch_sync (GCD) but I'll try to restrict the include

**Review by kambala-decapitator — APPROVED** [review]

**Pull request merged**